js/src/components/tree-select-control/options.js

```diff
@@ -83,9 +90,10 @@ const Options = ( {
 
 	return options.map( ( option ) => {
 		const isRoot = option.value === ROOT_VALUE;
-		const isExpanded = isRoot || nodesExpanded.includes( option.value );
 		const hasChildren = !! option.children?.length;
 		const optionIsChecked = isChecked( option );
+		const isExpanded =
+			filter.length || isRoot || nodesExpanded.includes( option.value );
```

Comment:
🚧 
Here needs typecasting for `filter.length`. Or, maybe change the `filter` prop to `isSearching` or `isFiltering`. Otherwise, it will set an invalid value to `aria-expanded` attribute.

![2022-03-15 18 08 11](https://user-images.githubusercontent.com/17420811/158355237-9b98a0a1-8fa2-4d6d-91ce-080df034e011.png)

Reply:
Insane catch! Thanks! Fixed